Library/Homebrew/livecheck/strategy/sparkle.rb

```diff
@@ -86,19 +86,29 @@ def self.items_from_content(content)
             enclosure = item.elements["enclosure"]
 
             if enclosure
-              url = enclosure["url"]
-              short_version = enclosure["shortVersionString"]
-              version = enclosure["version"]
-              os = enclosure["os"]
+              url = enclosure["url"].presence
+              short_version = enclosure["shortVersionString"].presence
+              version = enclosure["version"].presence
+              os = enclosure["os"].presence
             end
 
-            title = item.elements["title"]&.text&.strip
-            link = item.elements["link"]&.text&.strip
+            title = item.elements["title"]&.text&.strip&.presence
```

Comment:
That's a lot of `&`, a refactoring might be nice at some point given these are all doing the same thing (`&.text&.strip&.presence`)

Reply:
I'd been kind of thinking the same thing, even before adding the `#presence` calls. We could add a method for this in the `Xml` strategy:

```ruby
def self.element_text(parent, el_name)
  parent.get_elements(el_name)&.first&.text&.strip&.presence
end
```

The `#elements` method we're currently using works but it doesn't exist in the `REXML` docs, so this would also allow us to switch to `get_elements(...)&.first` without worrying about the added verbosity. [This switch would be necessary to get `brew typecheck` to pass anyway.]

Using this method, this code would end up like: `Xml.element_text(item, "title")`.

Anyway, this is just to say that I agree with the idea, it's an easy addition, and I can take care of it in a follow-up PR 👍

Reply:
> Using this method, this code would end up like: `Xml.element_text(item, "title")`.

This seems nicer.

> We could add a method for this in the `Xml` strategy

I'm not sure all the `&` are necessary here: better to do early `return` IMO in a method like that on the specific cases that may be `nil` rather than having to chain all of them. It's more code but a bit more explicit what's going on.

Reply:
> I'm not sure all the `&` are necessary here: better to do early `return`

I agree (the method above was just a rough idea). Using early `return`s, it looks something like this:

```ruby
def self.element_text(parent, el_name)
  element = parent.get_elements(el_name).first
  return if element.nil?

  text = element.text
  return if text.blank?

  text.strip
end
```

The first condition could also be `unless element`, `if element.blank?`, etc. but we can work out the fine details in a follow-up PR.

Reply:
Looks lovely 😍